datafusion/core/src/datasource/physical_plan/parquet/row_groups.rs

```diff
@@ -1003,6 +1006,246 @@ mod tests {
         );
     }
 
+    #[test]
+    fn row_group_pruning_predicate_utf8() {
```

Comment:
I believe the tests in this module are for row group pruning which use the statistics extraction code in 
https://github.com/apache/arrow-datafusion/blob/6c4109017edfe10800e0ffee8c1c254aade05849/datafusion/core/src/datasource/physical_plan/parquet/statistics.rs#L58-L57, which confusingly isn't the same code used to extract statistics for the entire file.

A way to test this might be to create a parquet exec to read `alltypes_plain.parquet'` and verify that statistics are present

For example, I think this information is encoded in the `physical_plan_with_stats` line like this

```
[(Col[0]:),(Col[1]:),(Col[2]:),(Col[3]:),(Col[4]:),(Col[5]:),(Col[6]:),(Col[7]:),(Col[8]:),(Col[9]:),(Col[10]:)]]     
```


```
❯ explain verbose select * from './parquet-testing/data/alltypes_plain.parquet';
....
| physical_plan_with_stats                                   | ParquetExec: file_groups={1 group: [[Users/andrewlamb/Software/arrow-datafusion/parquet-testing/data/alltypes_plain.parquet]]}, projection=[id, bool_col, tinyint_col, smallint_col, int_col, bigint_col, float_col, double_col, date_string_col, string_col, timestamp_col], statistics=[Rows=Exact(8), Bytes=Absent, [(Col[0]:),(Col[1]:),(Col[2]:),(Col[3]:),(Col[4]:),(Col[5]:),(Col[6]:),(Col[7]:),(Col[8]:),(Col[9]:),(Col[10]:)]]                                                                                                                                                                                                                              |
|                                                            |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
+------------------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+